src/olympia/addons/views.py

```diff
@@ -540,6 +540,13 @@ def icloud_bookmarks_redirect(request):
         return addon_detail(request, 'icloud-bookmarks')
 
 
+def find_replacement_addon(request):
+    guid = request.GET.get('guid')
+    if not guid:
+        raise http.Http404
```

Comment:
This would need to be customized eventually so it's not just generic, but we can cross that bridge later.